tests/unit/test_charm.py

```diff
@@ -338,132 +344,135 @@ def test_pro_airgapped_server_relation__success(self):
-        self.harness.enable_hooks()
+        self.start_leader_unit()
+
+        def update_snap_set_mock(expected_enabled: bool, expected_url: str):
+            def snap_set_mock(prefixed_configuration: Dict[str, Any]):
+                self.assertDictContainsSubset(
+                    {
+                        "lp.contracts.enabled": expected_enabled,
+                        "lp.contracts.url": expected_url,
+                    },
+                    prefixed_configuration,
+                )
+                self.assertNotIn("lp.contracts.user", prefixed_configuration)
+                self.assertNotIn("lp.contracts.password", prefixed_configuration)
+
+            self.snap_mock.set = Mock(side_effect=snap_set_mock)
 
+        update_snap_set_mock(True, "scheme://first.host:9999")
         pro_rel_id = self.harness.add_relation("pro-airgapped-server", "pro-airgapped-server")
         self.harness.add_relation_unit(pro_rel_id, "pro-airgapped-server/0")
         self.harness.update_relation_data(
             pro_rel_id,
             "pro-airgapped-server/0",
             {
-                "scheme": "http",
-                "hostname": "first.host.name",
+                "scheme": "scheme",
+                "hostname": "first.host",
+                "port": "9999",
             },
         )
+        self.assertTrue(self.snap_mock.set.called)
 
-        self.assertEqual(
-            self.harness.charm._state.pro_airgapped_address,
-            "http://first.host.name",
-        )
-
-        # Add another pro-airgapped-server unit.
+        # Adding another unit of `pro-airgapped-server`, but this new unit should not
+        # affect the Livepatch server configuration.
+        update_snap_set_mock(True, "scheme://first.host:9999")
         self.harness.add_relation_unit(pro_rel_id, "pro-airgapped-server/1")
         self.harness.update_relation_data(
             pro_rel_id,
             "pro-airgapped-server/1",
             {
-                "scheme": "http",
-                "hostname": "second.host.name",
+                "scheme": "scheme",
+                "hostname": "second.host",
+                "port": "9999",
             },
         )
-
-        # Since the first address is still available, the state should not change.
-        self.assertEqual(
-            self.harness.charm._state.pro_airgapped_address,
-            "http://first.host.name",
-        )
+        self.assertTrue(self.snap_mock.set.called)
 
     def test_pro_airgapped_server_relation__multiple_units_one_departs(self):
         """Test pro-airgapped-server relation when one of the relation units departs but the other one not."""
-        rel_id = self.harness.add_relation("livepatch", "livepatch")
-        self.harness.add_relation_unit(rel_id, f"{APP_NAME}/1")
-        self.harness.set_leader(True)
-        self.harness.enable_hooks()
+        self.start_leader_unit()
+
+        def update_snap_set_mock(expected_enabled: bool, expected_url: str):
```

Comment:
Interesting way of handling the test, I guess in the k8s charm this would be equivalent to just checking the pebble plan?

Reply:
Yeah, I think so.